### PR TITLE
api: create cache for device's JWT token

### PR DESCRIPTION
**Pull request by henrybarreto**

### CODE REVIEW

**Review by otavio — DISMISSED** [review]

File: api/services/auth.go [hunk 1]
```diff
@@ -44,11 +44,23 @@ func (s *service) AuthDevice(ctx context.Context, req *models.DeviceAuthRequest,
 		},
 	})
 
-	tokenStr, err := token.SignedString(s.privKey)
-	if err != nil {
+	var tokenStr string
+	if err := s.cache.Get(ctx, strings.Join([]string{"device", "token"}, "/"), &tokenStr); err != nil {
```
Comment (gustavosbarreto) [inline api/services/auth.go:48]: The key must be unique within devices fleet

File: api/services/auth.go [hunk 2]
```diff
@@ -33,6 +34,11 @@ type AuthService interface {
 }
 
 func (s *service) AuthDevice(ctx context.Context, req *models.DeviceAuthRequest, remoteAdrr string) (*models.DeviceAuthResponse, error) {
+	duration, err := rand.Int(rand.Reader, big.NewInt(5))
+	if err != nil {
+		return nil, err
+	}
+	time.Sleep(time.Second * time.Duration(duration.Int64()))
```
Comment (otavio) [inline api/services/auth.go:41]: Why? I don't see a reason to sleep here. Please provide a background reasining.

**Review by otavio — CHANGES_REQUESTED** [review]

File: api/services/auth.go (hunk 2 above)
Comment (henrybarreto) [reply api/services/auth.go:41]: This is a try to spread the auth ping's peak. I know, it seems to be a little dummy, but I've said, it is just a test.

I have forgotten to set this PR as a Draft.

**Comment by otavio** [comment]
@gustavosbarreto this seems as a dead PR; this isn't used nor need. I'll close it for now.

**Pull request closed without merging**